devices/tuya.js

```diff
@@ -288,6 +288,36 @@ const tzLocal = {
             }
         },
     },
+    // for HC-020-ZB, we must use different data points depending on thermostat_unit
+    hc020_temperatures: {
+        key: ['current_heating_setpoint', 'max_temperature', 'min_temperature', 'local_temperature_calibration'],
+        convertSet: async (entity, key, value, meta) => {
+            const fahrenheit = meta.state.thermostat_unit === 'fahrenheit' ? 1 : 0;
+            if (key === 'local_temperature_calibration') {
+                // when in fahrenheit mode, the thermostat does not report the re-calibrated value
+                // furthermore, the calibration value is applied in fahrenheit
+                // in this case, guess local_temperature based on the previous known calibration
+                // note that this will accumulate rounding errors if the calibration is changed too many times
+                const capped = utils.numberWithinRange(value, -9, 9);
+                await tuya.sendDataPointValue(entity, 27, capped);
+                const state = {local_temperature_calibration: value};
+                if (fahrenheit && meta.state.local_temperature != null && meta.state.local_temperature_calibration != null) {
+                    state['local_temperature'] = utils.precisionRound(meta.state.local_temperature +
+                        (capped - meta.state.local_temperature_calibration) * 5 / 9, 1);
```

Comment:
Unfortunate compromise here: the calibration unit changes with `thermostat_unit`! So if I want to keep `local_temperature` updated, I have to do an unfortunate rounding.

Another option is to toggle the `thermostat_unit`, which forces the thermostat to report the new temperature. Then we don't report strange fractional values. However, the thermostat will end up beeping *three* times...